server/src/main/java/org/elasticsearch/action/search/SearchTransportAPMMetrics.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.search;
+
+import org.elasticsearch.telemetry.TelemetryProvider;
+import org.elasticsearch.telemetry.metric.LongHistogram;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+public class SearchTransportAPMMetrics {
+    private final LongHistogram actionLatencies;
+
+    public SearchTransportAPMMetrics(MeterRegistry meterRegistry) {
+        this(
+            meterRegistry.registerLongHistogram(
+                "es.transport.action.latency.histogram",
+                "Transport action execution times expressed as a histogram",
+                "millis"
+            )
```

Comment:
A histogram metric type dynamically creates buckets based on the data it receives. 

Histograms (e.g. percentiles) must be calculated at the point of collection as opposed to trying to create one in Kibana that is derived from a single counter metric. This is because timing data is discarded when using a single counter, as all recorded measurements are stored in an single Atomic Long (or equivalent), and timing is only ascribed when the metrics are polled by the collection pipeline.  Thus, if we need that timing granularity to accurately produce percentiles, we must record it in a histogram at the point of collection.